src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -130,13 +130,15 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
             );
 
             let mut is_loop_move = false;
+            let mut is_partial_move = false;
             for move_site in &move_site_vec {
                 let move_out = self.move_data.moves[(*move_site).moi];
                 let moved_place = &self.move_data.move_paths[move_out.path].place;
 
                 let move_spans = self.move_spans(moved_place, move_out.source);
                 let move_span = move_spans.args_or_use();
 
+                is_partial_move = used_place.is_prefix_of(moved_place);
```

Comment:
We should probably check that we're not setting this back to `false` at any point, perhaps just assign directly into `is_partial_move` with `move_site_vec.iter().any(|move_site| { /* condition */ })`.

Reply:
Good idea. The downside with the new way I've implemented this is that is certainly less efficient. I'm not sure how much that matters here. We could `is_prefix || ...` in the for loop, I guess. The `iter().any` is certainly clearer though...

Reply:
I don't think it would be so much less efficient that it is worth worrying about. Feel free to take either approach.